### ISSUE TITLE
Diff view ignores .gitattributes settings

### ISSUE THREAD

**Issue by gardiner** [rowanj/gitx#170]
**Steps to reproduce:**

Add some file definitions to .gitattributes to mark them as binary, e.g.:

`*.pdf binary`

**Expected behavior:**

The files should be excluded them from the diff view.

**Actual behavior:**

The files are still being displayed including a full diff.

**Details:**

Sometimes git/GitX interprets specific resources (xml files, pdf files) as text files and tries to display diffs when these files have been changed. That makes the diff view hard to use.
With command-line git these files can be easily excluded from diffs by adding them to a .gitattributes file. This file seems to be ignored by GitX.

**Comment by rowanj** [comment]
Indeed, I think the diff system explicitly gets the diff for each file, which bypasses the attribute handling stuff.

**Comment by robbarrett** [comment]
I'm having the opposite problem, which I thought I'd add to this.

I've added lines to .gitattributes and .git/config to enable git to generate textual diffs of binary MSWord files (cf. http://git-scm.com/book/ch7-2.html).  This works fine from the command line, but GitX's diff reports "Binary file differs".

Would it be possible to handle the diff the same way git itself does?
